### ISSUE TITLE
Articles marked previously as read (with shift + A) are downloading from Nextcloud as unread.

### ISSUE THREAD

**Issue by JakubMajcher** [newsboat/newsboat#2166]
**Newsboat version (copy the output of `newsboat -v` or the first line of `git show`)**:
```newsboat 2.21.0 - https://newsboat.org/
Copyright (C) 2006-2015 Andreas Krennmair
Copyright (C) 2015-2020 Alexander Batischev
Copyright (C) 2006-2017 Newsbeuter contributors
Copyright (C) 2017-2020 Newsboat contributors

Newsboat jest wolnym oprogramowaniem na licencji MIT License (Użyj `newsboat -vv' aby zobaczyć pełny tekst licencji.)
Zawiera:
- Zawiera bibliotekę JSON for Modern C++ na licencji MIT License: https://github.com/nlohmann/json
- Zawiera bibliotekę  optional-lite, na licencji the Boost Software License: https://github.com/martinmoene/optional-lite
- Zawiera bibliotekę  expected-lite, na licencji the Boost Software License: https://github.com/martinmoene/expected-lite

newsboat 2.21.0
System: Linux 5.18.0-4-amd64 (x86_64)
Compiler: g++ 11.3.0
ncurses: ncurses 6.3.20220423 (compiled with 6.3)
libcurl: libcurl/7.84.0 GnuTLS/3.7.7 zlib/1.2.11 brotli/1.0.9 zstd/1.5.2 libidn2/2.3.3 libpsl/0.21.0 (+libidn2/2.3.0) libssh2/1.10.0 nghttp2/1.48.0 librtmp/2.3 OpenLDAP/2.5.12 (compiled with 7.84.0)
SQLite: 3.39.2 (compiled with 3.39.1)
libxml2: compiled with 2.9.14
```

Config file (copy from ~/.newsboat/config or ~/.config/newsboat/config):

```
# Nextcloud configuration

urls-source "ocnews"
ocnews-url "http://192.168.1.250:1234"
ocnews-login "kuba"
ocnews-password "z8ms5-E6MTc-TnKsY-wdNrG-TJBcL"
ssl-verifyhost "no"

##############

# How to display feeds and articles

feed-sort-order firsttag
show-read-feeds no
show-read-articles no
article-sort-order date-desc

##############

# Other configuration

confirm-exit yes
unbind-key d
unbind-key ^D

##############

# External browser

browser "~/.mozilla2/firefox2/firefox-bin --new-tab %u"

##############

# Macros

macro w set browser "~/.newsboat/ownscripts/wallabag %u" ; open-in-browser-and-mark-read ; set browser "~/.mozilla2/firefox2/firefox-bin --new-tab %u"
macro y set browser "~/.newsboat/ownscripts/youtube %u" ; open-in-browser-and-mark-read ; set browser "~/.mozilla2/firefox2/firefox-bin --new-tab %u"
macro m set browser "~/.newsboat/ownscripts/mp3 %u" ; open-in-browser-and-mark-read ; set browser "~/.mozilla2/firefox2/firefox-bin --new-tab %u"
macro h set browser "~/.newsboat/ownscripts/mp3-hr %u" ; open-in-browser-and-mark-read ; set browser "~/.mozilla2/firefox2/firefox-bin --new-tab %u"

##############

# Theme

color info 					default default reverse
color listnormal_unread 	cyan default
color listfocus 			blue default reverse bold
color listfocus_unread 		blue default reverse bold
color article 				cyan default
color listnormal 			yellow default

text-width 80

datetime-format "%d.%m.%Y, %R"
feedlist-format "%-8u | %-20T | %t"
articlelist-format "%-1f | %D | %?T?|%-17T| ?%-100t | %a"

feedlist-title-format "%?F?Feeds&Your feeds? (%u unread, %t total)%?F? matching filter '%F'&?%?T? - tag '%T'&?" 
filebrowser-title-format "%?O?Open File&Save File? - %f"
itemview-title-format "Article '%T' (%u unread, %t total)" 
searchresult-title-format "Search results for '%s' (%u unread, %t total)%?F? matching filter '%F'&?" 
selectfilter-title-format "Select Filter"
selecttag-title-format "Select Tag"
urlview-title-format "URLs"
articlelist-title-format "Articles in feed '%T' (%u unread, %t total)%?F? matching filter `%F'&? - %U"
dialogs-title-format "Dialogs"

highlight feedlist    "^  *[0-9]+  *N  "                    magenta  magenta
highlight articlelist "^  *[0-9]+  *N  "                    magenta  magenta

highlight article     "(^Kanał:.*|^Tytuł:.*|^Autor:.*)"     red      default
highlight article     "(^Odsyłacz:.*|^Data:.*)"             white    default
highlight article     "^Podcast Download URL:.*"            cyan     default
highlight article     "^Odsyłacze:"                         magenta  black    underline
highlight article     "https?://[^ ]+"                      green    default
highlight article     "^(Title):.*$"                        blue     default
highlight article     "\\[[0-9][0-9]*\\]"                   magenta  default  bold
highlight article     "\\[image\\ [0-9]+\\]"                green    default  bold
highlight article     "\\[embedded flash: [0-9][0-9]*\\]"   green    default  bold
highlight article     ":.*\\(link\\)$"                      cyan     default
highlight article     ":.*\\(image\\)$"                     blue     default
highlight article     ":.*\\(embedded flash\\)$"            magenta  default
```

Steps to reproduce the issue:

1. Run app with Nextcloud

2. Go inside feed (ENTER)

3. Mark all as read (shift + A)

4. Refresh (shift + R)

5. Articles marked previously as read are downloaded from Nextcloud as unread.

**Other info you think is relevant**:
Doesn't happen when we mark article as read when we go into article.

**Comment by Minoru** [comment]
What version of Nextcloud News are you running? Apparently there was a bug that was fixed in 18.0.1-beta2 on 2022-03-22: https://github.com/nextcloud/news/blob/master/CHANGELOG.md#1801-beta2---2022-03-22

**Comment by JakubMajcher** [comment]
@Minoru what version? Really old one. :D 15.2.0. I need to upgrade my nextcloud instance and then I'll give You some info here. Anyway, last year I used few another rss clients and I didn't have problem like this. But I guess this clients didn't use this API.
